mods/creative/inventory.lua

```diff
@@ -194,10 +194,29 @@ function creative.register_tab(name, title, items)
 	})
 end
 
+-- Sort registered items
+local registered_nodes = {}
+local registered_tools = {}
+local registered_craftitems = {}
+
+minetest.register_on_mods_loaded(function()
+	for name, def in pairs(minetest.registered_items) do
+		local group = def.groups or {}
+
+		if group.node or minetest.registered_nodes[name] then
+			registered_nodes[name] = def
+		elseif group.tool or minetest.registered_tools[name] then
+			registered_tools[name] = def
+		elseif group.craftitem or minetest.registered_craftitems[name] then
+			registered_craftitems[name] = def
+		end
```

Comment:
I'm sorry for being too late, but this won't work if you want to move eg. a node into the craftitem category.